### PR TITLE
Allow training progress bars to be disabled

### LINKED ISSUE #217: Add option to disable the progress bar
Hi,

unless I missed something, it is currently not possible to disable the progress bar when training a SetFit model using `SetFitTrainer`. I suggest adding an option for this.

**Motivation**: I am using SetFit [in a Jupyter Notebook](https://github.com/webis-de/small-text/blob/5b5c5b6da1bdab44d0ff5974e177fc81e46f8f46/examples/notebooks/03-active-learning-with-setfit.ipynb) (not sure why it seems to print more than in the SetFit example notebook) and the output makes it rather ugly. Moreover, one might want not to display progress bars regardless of the notebook setting. 

**Problematic calls**:
https://github.com/huggingface/setfit/blob/fa1021d2355f0cb3a2c85732ee7ffe44b0cef0d1/src/setfit/trainer.py#L374
https://github.com/huggingface/setfit/blob/fa1021d2355f0cb3a2c85732ee7ffe44b0cef0d1/src/setfit/trainer.py#L388

### PR DESCRIPTION
**Pull request by tomaarsen**
Resolves #217

Hello!

## Pull request overview
* Tackle #217
* Normalize docstring type hints throughout the project

## Details
I've simply exposed a `show_progress_bar` argument on both the `train` method for both `SetFitTrainer` and `DistillationSetFitTrainer`, which passes the parameter directly to the `fit` methods of the body (a sentence transformer) and the full model, which causes it to train the current head. `show_progress_bar` was already an argument for `SetFitModel.fit`, so few changes were needed.

### Additionally...
I also normalized some docstring type hints. The train method was the only method where the types mentioned in the docstring were not encompassed by backticks. I've resolved that now via aecae75e4b0096f5de20ead429d9b818fc82fd74.

## Quick testing scripts

<details><summary>Testing script using LogisticRegression head</summary>

```python
from datasets import load_dataset
from sentence_transformers.losses import CosineSimilarityLoss

from setfit import SetFitModel, SetFitTrainer, sample_dataset


# Load a dataset from the Hugging Face Hub
dataset = load_dataset("sst2")

# Simulate the few-shot regime by sampling 8 examples per class
train_dataset = sample_dataset(dataset["train"], label_column="label", num_samples=8)
eval_dataset = dataset["validation"]

# Initialize `SetFitModel` with the differentiable head
model = SetFitModel.from_pretrained(
    "sentence-transformers/paraphrase-albert-small-v2",
)

# Create trainer
trainer = SetFitTrainer(
    model=model,
    train_dataset=train_dataset,
    eval_dataset=eval_dataset,
    loss_class=CosineSimilarityLoss,
    metric="accuracy",
    batch_size=16,
    num_iterations=20,  # The number of text pairs to generate for contrastive learning
    num_epochs=1,  # The number of epochs to use for constrastive learning
    column_mapping={"sentence": "text", "label": "label"}  # Map dataset columns to text/label expected by trainer
)

trainer.train(show_progress_bar=False)

metrics = trainer.evaluate()
print(metrics)
```
</details>

<details><summary>Testing script using differentiable head</summary>

```python
from datasets import load_dataset
from sentence_transformers.losses import CosineSimilarityLoss

from setfit import SetFitModel, SetFitTrainer, sample_dataset


# Load a dataset from the Hugging Face Hub
dataset = load_dataset("sst2")

# Simulate the few-shot regime by sampling 8 examples per class
num_classes = 2
train_dataset = sample_dataset(dataset["train"], label_column="label", num_samples=8)
eval_dataset = dataset["validation"]

# Initialize `SetFitModel` with the differentiable head
model = SetFitModel.from_pretrained(
    "sentence-transformers/paraphrase-albert-small-v2",
    use_differentiable_head=True,
    head_params={"out_features": num_classes},
)

# Create trainer
trainer = SetFitTrainer(
    model=model,
    train_dataset=train_dataset,
    eval_dataset=eval_dataset,
    loss_class=CosineSimilarityLoss,
    metric="accuracy",
    batch_size=16,
    num_iterations=20,  # The number of text pairs to generate for contrastive learning
    num_epochs=1,  # The number of epochs to use for constrastive learning
    column_mapping={"sentence": "text", "label": "label"}  # Map dataset columns to text/label expected by trainer
)

# Freeze head
trainer.freeze()

# Do contrastive training
trainer.train(show_progress_bar=False)

# Unfreeze head
trainer.unfreeze()

# Unfreeze head and freeze body
trainer.unfreeze(keep_body_frozen=True)

# Train end-to-end
trainer.train(num_epochs=3, show_progress_bar=False)

metrics = trainer.evaluate()
print(metrics)
```
</details>

- Tom Aarsen

### CODE REVIEW

**Review by lewtun — APPROVED** [review]
Thanks for the improvement @tomaarsen - this looks great to me!

We just have some merge conflicts that need to be resolved before merging :)
